server/src/main/java/io/crate/metadata/information/InformationColumnsTableInfo.java

```diff
@@ -81,7 +82,18 @@
                     return null;
                 }
             })
-            .add("character_maximum_length", INTEGER, ignored -> null)
+            .add("character_maximum_length", INTEGER, r -> {
+                if (DataTypes.isSameType(r.info.valueType(), STRING)) {
```

Comment:
```suggestion
                if (r.info.valueType() instanceof StringType) {
```

Given that there is a cast in the next line, the `isSameType` is kinda unnecessary abstraction here and it would also imply that this callsite is aware about the implementation details of `isSameType`